project-addons/custom_partner/partner.py

```diff
@@ -51,53 +54,113 @@ def _calculate_annual_invoiced(self):
         partner_ids = partner_obj.search([('is_company', '=', True), ('child_ids', '!=', False)])
         actual_year = datetime.now().year
         actual_month = datetime.now().month
+        past_month = actual_month - 1
         actual_day = datetime.now().day
         start_year = str(actual_year) + '-01-01'
+        start_past_year = str(actual_year - 1) + '-01-01'
         start_month = str(actual_year) + '-' + str(actual_month) + '-01'
+        start_past_month = str(actual_year) + '-' + str(past_month) + '-01'
         end_year = str(actual_year) + '-12-31'
+        end_past_year = str(actual_year - 1) + '-12-31'
         end_month = str(actual_year) + '-' + str(actual_month) + '-' + str(actual_day)
+        end_day_past_month = monthrange(actual_year, past_month)
+        end_past_month = str(actual_year) + '-' + str(past_month) + '-' + str(end_day_past_month[1])
         for partner in partner_ids:
             invoice_ids_year = invoice_obj.search([('date_invoice', '>=', start_year),
-                                              ('date_invoice', '<=', end_year),
-                                              '|',
-                                              ('state', '=', 'open'),
-                                              ('state', '=', 'paid'),
-                                              '|',
-                                              ('partner_id', '=', partner.id),
-                                              ('partner_id.parent_id', '=', partner.id)])
+                                                   ('date_invoice', '<=', end_year),
+                                                   '|',
+                                                   ('state', '=', 'open'),
+                                                   ('state', '=', 'paid'),
+                                                   '|',
+                                                   ('partner_id', '=', partner.id),
+                                                   ('partner_id.parent_id', '=', partner.id)])
+
+            invoice_ids_past_year = invoice_obj.search([('date_invoice', '>=', start_past_year),
+                                                        ('date_invoice', '<=', end_past_year),
+                                                        '|',
+                                                        ('state', '=', 'open'),
+                                                        ('state', '=', 'paid'),
+                                                        '|',
+                                                        ('partner_id', '=', partner.id),
+                                                        ('partner_id.parent_id', '=', partner.id)])
```

Comment:
Tampoco estáis filtrando facturas de compra y en cambio abajo suponéis que son abonos o ventas, en la mayoría de los casos puede ser así, peor es posible que haya algún cliente que sea cliente y proveedor o que lo haya en el futuro